File: OpenCV/06_filtrado_convolucion/cpp/main.cpp
#include <opencv2/opencv.hpp>
int main(){ auto im=cv::imread("input.jpg"); if(im.empty()) return 1; cv::Mat a,b,c; cv::GaussianBlur(im,a,{7,7},1.5); cv::medianBlur(im,b,7); cv::bilateralFilter(im,c,9,75,75); cv::imwrite("output_gaussian.png",a); cv::imwrite("output_median.png",b); cv::imwrite("output_bilateral.png",c); }

File: OpenCV/06_filtrado_convolucion/python/main.py
import cv2
img=cv2.imread("input.jpg"); assert img is not None
gauss=cv2.GaussianBlur(img,(7,7),1.5); median=cv2.medianBlur(img,7); bilateral=cv2.bilateralFilter(img,9,75,75)
cv2.imwrite("output_gaussian.png",gauss); cv2.imwrite("output_median.png",median); cv2.imwrite("output_bilateral.png",bilateral)
